pgxn/neon/pagestore_client.h

```diff
@@ -298,4 +298,46 @@ extern void set_cached_relsize(NRelFileInfo rinfo, ForkNumber forknum, BlockNumb
 extern void update_cached_relsize(NRelFileInfo rinfo, ForkNumber forknum, BlockNumber size);
 extern void forget_cached_relsize(NRelFileInfo rinfo, ForkNumber forknum);
 
+/*
+ * Relation kind enum.
+ */
+typedef enum
+{
+    /* The persistence is not known */
+	RELKIND_UNKNOWN,
+
+    /* The relation is a permanent relation that is WAL-logged normally */
+	RELKIND_PERMANENT,
+
+    /* The relation is an unlogged table/index, stored only on local disk */
+	RELKIND_UNLOGGED,
+
+    /*
+     * The relation is a permanent (index) relation, but it is being built by an in-progress
+     * transaction. It currently only lives on local disk and hasn't been WAL-logged yet.
+     * It will turn into a permanent relation later when the index build completes.
+     * This is currently used for GiST, SP-GiST and GIN indexes, as well as the pgvector
+     * extension.
+     */
+	RELKIND_UNLOGGED_BUILD
+} RelKind;
+
+/* entry type stored in relkind_hash */
+typedef struct
```

Comment:
It'd be worth noting somewhere that we have just one entry for the whole relation, i.e. we don't have separate entries for the individual forks. It gets a little complicated with unlogged relations. The main fork of an unlogged relation is considered UNLOGGED, but its init-fork is treated as PERMANENT.